code/datums/perks/perk.dm

```diff
@@ -80,3 +80,11 @@
 
 /datum/perk/proc/is_active()
 	return active
+
+/mob/proc/has_perk(perk_type)
+	if(!stats)
+		return FALSE
+	for(var/datum/perk/P in stats.perks)
+		if(P.type == perk_type)
```

Comment:
You should use `stats.getPerk(perk_type)` or at least `locate(perk_type) in stats.perks`.